common/resource/fx.go

```diff
@@ -349,7 +350,14 @@ func HistoryClientProvider(clientBean client.Bean) historyservice.HistoryService
 	historyClient := history.NewRetryableClient(
 		historyRawClient,
 		common.CreateHistoryClientRetryPolicy(),
-		common.IsServiceClientTransientError,
+		func(err error) bool {
+			if err.Error() == interceptor.ErrNamespaceHandover.Error() {
```

Comment:
I guess here I just want to start the conversation around which layer should retry this error. So seems like you are suggesting no retry should be done for this error in our system? 

Current behavior is frontend handler & frontend client (used by worker role I believe) will retry this error.

Reply:
hmmm, my initial thought was we should not retry. But now I think we should because the namespace in handover state is a transient error. Once handover is done, it can forward to new active cluster.
Doing retry will reduce error returned to user.

Reply:
Yeah exactly, that why I still want to retry on frontend (handler or client), but not for history client & handler (updated in this PR).

Actually, the current behavior is there's no retry for this error on frontend handler as the interceptor for checking namespace's replication state if before the retry interceptor. Let's probably sync offline on this.

Reply:
Synced offline. We decided to keep the existing behavior and do not retry this error (in frontend handler, history client/handler, etc.)

Reply:
Moved this check to `IsServiceHandlerRetryableError` which is called by `IsServiceClientTransientError`. So now neither client or handler will retry this ns handover error. 

But due to some cycle dependency issue (between common/rpc/interceptor and common), I also moved the error definition to common package.